v2/kafka-to-gcs/src/main/java/com/google/cloud/teleport/v2/transforms/AvroWriteTransform.java

```diff
@@ -196,19 +250,26 @@ public WriteFilesResult<AvroDestination> writeToGCS(PCollection<GenericRecord> g
   }
 
   public static class ConvertBytesToGenericRecord
-      extends DoFn<KafkaRecord<byte[], byte[]>, GenericRecord> {
+      extends DoFn<
+          KafkaRecord<byte[], byte[]>,
+          FailsafeElement<KafkaRecord<byte[], byte[]>, GenericRecord>> {
     private transient SchemaRegistryClient schemaRegistryClient;
     private String schemaRegistryURL = null;
     private String schemaPath = null;
     private boolean useMock;
 
-    public ConvertBytesToGenericRecord(String schemaRegistryURL) {
+    private final BadRecordRouter badRecordRouter;
+
+    public ConvertBytesToGenericRecord(String schemaRegistryURL, BadRecordRouter badRecordRouter) {
       this.schemaRegistryURL = schemaRegistryURL;
+      this.badRecordRouter = badRecordRouter;
     }
 
-    public ConvertBytesToGenericRecord(String schemaPath, boolean useMock) {
+    public ConvertBytesToGenericRecord(
+        String schemaPath, boolean useMock, BadRecordRouter badRecordRouter) {
       this.schemaPath = schemaPath;
       this.useMock = useMock;
+      this.badRecordRouter = badRecordRouter;
     }
 
     @DoFn.Setup
```

Comment:
This `@Setup` implementation will result in the schema being loaded from GCS every time a new DoFn instance is constructed (could be hundreds/thousands times). I suggest loading schema once in the template code, when the pipeline is constructed, and passing the loaded schema string to the DoFn.

Reply:
Added a TODO.